salt/_modules/metalk8s_package_manager.py

```diff
@@ -0,0 +1,78 @@
+'''
+Describes our custom way to deal with yum packages
+so that we can support downgrade in metalk8s
+'''
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'metalk8s_package_manager'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def list_pkg_deps(pkg_name, version=None, fromrepo=None):
+    '''
+    Check dependencies related to the packages installed so that we can pass
+    this information to pkg.installed
+
+    name
+        Name of the package installed
+
+    version
+        Version number of the package
+
+        Use : salt '*' metalk8s_package_manager.list_pkg_deps kubelet 1.11.9
+    '''
+    log.info(
+        'Listing deps for "%s" with version "%s"',
+        str(pkg_name),
+        str(version)
+    )
+    pkgs_list = {pkg_name: version}
+
+    if not version:
+        return pkgs_list
+
+    command_all = [
+        'repoquery', '--whatrequires', '--qf',
+        '%{NAME} %{VERSION}-%{RELEASE}',
+        '{}-{}'.format(str(pkg_name), str(version))
+    ]
+
+    if fromrepo:
+        command_all.append('--disablerepo="*"')
+        command_all.append('--enablerepo="{}"'.format(fromrepo))
+
+    deps_list = __salt__['cmd.run_all'](command_all)
+
+    if deps_list['retcode'] != 0:
+        log.error(
+            'Failed to list package dependencies: %s',
+            deps_list['stderr'] or deps_list['stdout']
+        )
+        return None
+
+    out = deps_list['stdout'].splitlines()
+    for line in out:
+        name, version = line.strip().split()
```

Comment:
Would be nice to have some error management, with some `try:`/`except ValueError:` in case some lines are not in the expected format. (minor, since you can rely on `repoquery` to have a constant output format).